SMLHelper/Options/ModConfig.cs

```diff
@@ -0,0 +1,226 @@
+        /// <param name="saveDefaultConfigIfNotExist">Whether a config file containing default values should
+        /// be created if it does not already exist.</param>
+        /// <returns>A <seealso cref="ModConfig"/> with its properties and fields populated from the 
+        /// associated JSON file.</returns>
+        /// <example>
+        /// <code>
+        /// using System;
+        /// using SMLHelper.V2.Options;
+        /// using SMLHelper.V2.Utility;
+        /// using UnityEngine;
+        /// 
+        /// public class Config : ModConfig {
+        ///     public KeyCode ActivationKey { get; set; } = KeyCode.Backspace;
+        ///     public Config() : base(fileName: "config") { }
+        /// }
+        /// 
+        /// public class MyMod {
+        ///     public static Config Config = ModConfig.Load(new Config());
+        ///     public static void Initialize() {
+        ///         string activationKey = KeyCodeUtils.KeyCodeToString(Config.ActivationKey);
+        ///         Console.WriteLine($"[MyMod] LOADED: ActivationKey = {activationKey}");
+        ///         // Will print "[MyMod] LOADED: ActivationKey = Backspace" in Player.log on first run,
+        ///         // and whatever value they have saved in their config.json for the "ActivationKey"
+        ///         // property on subsequent runs.
+        ///     }
+        /// }
+        /// </code>
+        /// </example>
+        /// <seealso cref="Load{T}(bool)"/>
+        /// <seealso cref="Load(bool)"/>
+        /// <seealso cref="Save"/>
+        public static T Load<T>(T config, bool saveDefaultConfigIfNotExist = true) where T : ModConfig
+        {
+            var path = config.ConfigPath;
+            if (Directory.Exists(Path.GetDirectoryName(path)) && File.Exists(path))
+            {
+                try
+                {
+                    string serializedJSON = File.ReadAllText(path);
+                    config = JsonConvert.DeserializeObject<T>(serializedJSON,
+                        new JsonConverters.KeyCodeConverter());
+                }
+                catch (Exception ex)
+                {   // Ideally, what we want is to pop up a QMM Dialog letting the user know exactly what mod
+                    // and exactly what file is causing the error, with an option to open a JSON validator in
+                    // browser to help them fix the problem.
+                    // We'd also want this issue to stop QMods from considering the mod loaded properly, or
+                    // expose a method for mods to check if any mods had config file errors, so that mods
+                    // which rely on all mods loading properly before performing actions can act accordingly.
+                    // For now, without copying QMM's dialog class into SML, this is the best we can do.
+                    Logger.Announce($"Could not parse JSON file: {path}", LogLevel.Error, true);
+                    throw ex;
```

Comment:
Honestly, I wouldn't bother.

If something is wrong with the config file as read, I would simply replace it with the default config instead.  
Sure, we should log the error, but I wouldn't stop the the mod from loading over that.

If this seems to heavy handed, you could make it another option (defaulting to _yes_).

Reply:
I'm honestly happy to have it just load the defaults, have been wanting input on this.

Reply:
Using the default config whenever there is a serialization exception is what I (and I'm sure plenty others) have been doing for a while. :+1: